backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -409,6 +414,82 @@ public FutureExperimentRunDAO(
+    for (Artifact dataset : newDatasets) {
+      String datasetVersionId = dataset.getLinkedArtifactId();
+      if (!datasetVersionId.isEmpty()) {
+        newDatasetLinkedIds.add(datasetVersionId);
+      }
+    }
+
+    if (newDatasetLinkedIds.isEmpty()) {
+      return InternalFuture.completedInternalFuture(newDatasets);
+    }
+    return jdbi.withHandle(
+            handle -> {
+              Map<String, Long> datasetVersionDatasetMap = new HashMap<>();
+              handle
+                  .createQuery(
+                      " SELECT commit_hash, repository_id FROM repository_commit WHERE commit_hash IN (<datasetIds>) ")
+                  .bindList("datasetIds", newDatasetLinkedIds)
+                  .map(
+                      (rs, ctx) ->
+                          datasetVersionDatasetMap.put(
+                              rs.getString("commit_hash"), rs.getLong("repository_id")))
+                  .list();
+              return datasetVersionDatasetMap;
+            })
+        .thenCompose(
+            datasetVersionDatasetMap -> {
+              List<String> accessibleDatasetVersionIds = new ArrayList<>();
+              List<InternalFuture<Artifact>> internalFutures = new LinkedList<>();
+              for (Artifact dataset : newDatasets) {
+                String datasetVersionId = dataset.getLinkedArtifactId();
+                if (!datasetVersionId.isEmpty()
+                    && !accessibleDatasetVersionIds.contains(datasetVersionId)) {
+                  try {
+                    if (datasetVersionDatasetMap.containsKey(datasetVersionId)
+                        && datasetVersionDatasetMap.get(datasetVersionId) != null) {
+                      Long datasetId = datasetVersionDatasetMap.get(datasetVersionId);
+
+                      internalFutures.add(
+                          checkEntityPermissionBasedOnResourceTypes(
+                                  Collections.singletonList(String.valueOf(datasetId)),
+                                  ModelDBActionEnum.ModelDBServiceActions.READ,
+                                  ModelDBServiceResourceTypes.DATASET)
+                              .thenCompose(
+                                  unused -> {
+                                    accessibleDatasetVersionIds.add(datasetVersionId);
+                                    return InternalFuture.completedInternalFuture(dataset);
+                                  },
+                                  executor));
+                    } else {
+                      return InternalFuture.failedStage(
+                          new InvalidArgumentException(
+                              "Dataset not found for dataset version: " + datasetVersionId));
+                    }
+                  } catch (Exception ex) {
+                    LOGGER.trace(ex.getMessage());
+                    if (errorOut) {
+                      return InternalFuture.failedStage(ex);
+                    }
+                  }
```

Comment:
This is wrong as we shouldn't just ignore legit errors. For example, if UAC is unavailable, we should still fail regardless of what `Exception` says.

Reply:
Catch only permission denied exception